R/tbl-lazy.R

```diff
@@ -201,7 +201,14 @@ distinct_.tbl_lazy <- function(.data, ..., .dots = list(), .keep_all = FALSE) {
 add_op_join <- function(x, y, type, by = NULL, copy = FALSE,
                         suffix = c(".x", ".y"),
                         auto_index = FALSE, ...) {
-  by <- common_by(by, x, y)
+  by_intersect <- intersect(tbl_vars(x), tbl_vars(y))
```

Comment:
Thanks; https://github.com/tidyverse/dbplyr/pull/24/commits/6e5daa9bb7293b074c540946df1810ef46e59839 fixes it